modCrypto/src/org/aion/crypto/AddressSpecs.java

```diff
@@ -25,4 +26,26 @@ private AddressSpecs() {}
         buf.put(HashUtil.h256(publicKey), 1, 31);
         return buf.array();
     }
+
+    public static String checksummedAddress(String address) {
+        assert address != null;
+        address = address.replace("0x", "");
```

Comment:
If we are given a string with "0x" sitting in the middle of it, or sitting in multiple places, this replacement could strip the string back down to a legitimate address - do we want this kind of behaviour?

Reply:
I think this is an interesting question, though I don't have a clear idea of the gravity of the implications.

I assume it's easy enough to "fix", simply only replace starting "0x"s?

Reply:
I think the "0x not at the start of string" problem still needs to be fixed (also can yuo add a unit test to make sure this doesn't happen?)

Suggestion: `address.replaceFirst("^0x", "")`

https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replaceFirst(java.lang.String,%20java.lang.String)